examples/sparse_gamma_def.py

```diff
@@ -126,13 +132,48 @@ def sample_ws(name, width):
             sample_zs("mid", self.mid_width)
             sample_zs("bottom", self.bottom_width)
 
-    # define a helper function to clip parameters defining the custom guide.
-    # (this is to avoid regions of the gamma distributions with extremely small means)
-    def clip_params(self):
-        for param, clip in zip(("log_alpha", "log_mean"), (-2.5, -4.5)):
-            for layer in ["top", "mid", "bottom"]:
-                for wz in ["_w_q_", "_z_q_"]:
-                    pyro.param(param + wz + layer).data.clamp_(min=clip)
+
+# define a helper function to clip parameters defining the custom guide.
+# (this is to avoid regions of the gamma distributions with extremely small means)
+def clip_params():
+    for param, clip in zip(("alpha", "mean"), (-2.5, -4.5)):
+        for layer in ["_q_top", "_q_mid", "_q_bottom"]:
+            for wz in ["_w", "_z"]:
+                pyro.param(param + wz + layer).data.clamp_(min=clip)
+
+
+# Define a custom guide using the EasyGuide class.
+# Unlike the 'auto' guide, this guide supports data subsampling.
+# This is the best performing of the three guides.
+# (This guide is functionally similar to the auto guide, but performs
+# somewhat better. The reason seems to be some combination of: i) the better
+# numerical stability of the softplus; and ii) the custom initialization.
+# Though note that in the easy guide case KL divergences are not computed
```

Comment:
I believe the cause of `TraceMeanField_ELBO`'s missed opportunity to use KL is due to `EasyGuide`'s pattern of using `Delta` distributions to split up a single `Normal` distribution; this is equally a problem for `AutoGuide`. Could you update the comment to reflect this?
> Note that in the easy guide and auto guide cases, KL divergences are not computed analytically in the ELBO because the guide and model sample sites are separated by a Delta distribution for reshaping.

Reply:
done. lmk if not sufficient